scripts/prepare_sector_network.py

```diff
@@ -2543,6 +2543,27 @@ def add_biomass(n, costs):
             marginal_cost=costs.at["BioSNG", "efficiency"] * costs.at["BioSNG", "VOM"],
         )
 
+    if options["bioH2"]:
+        name = (pd.Index(spatial.biomass.nodes) + " "
+                + pd.Index(spatial.h2.nodes.str.replace(" H2", "")))
+        n.madd(
+            "Link",
+            name,
+            suffix=" solid biomass to hydrogen CC",
+            bus0=spatial.biomass.nodes,
+            bus1=spatial.h2.nodes,
+            bus2=spatial.co2.nodes,
+            bus3="co2 atmosphere",
+            carrier="solid biomass to hydrogen",
+            efficiency=costs.at['solid biomass to hydrogen', 'efficiency'],
+            efficiency2=costs.at['solid biomass', 'CO2 intensity'] * options["cc_fraction"],
+            efficiency3=-costs.at['solid biomass', 'CO2 intensity'] * options["cc_fraction"],
+            p_nom_extendable=True,
+            capital_cost=costs.at['solid biomass to hydrogen', 'fixed'] * costs.at['solid biomass to hydrogen', 'efficiency']
```

Comment:
Just to double-check, the unit in https://github.com/PyPSA/technology-data/blob/523eea9f36867da8fcf5e1e2aa91f1d4fe9a667b/outputs/costs_2030.csv#L1026 is €/kW_th H2?

Reply:
just remembered that we now always need to report the investments as well, i.e. 
```
overnight_cost=costs.at['solid biomass to hydrogen', 'investment'] * costs.at['solid biomass to hydrogen', 'efficiency'] + costs.at['biomass CHP capture', 'investment'] * costs.at['solid biomass', 'CO2 intensity'],

```

Reply:
> Just to double-check, the unit in https://github.com/PyPSA/technology-data/blob/523eea9f36867da8fcf5e1e2aa91f1d4fe9a667b/outputs/costs_2030.csv#L1026 is €/kW_th H2?

Yes:
![image](https://github.com/user-attachments/assets/6e634082-3655-44ec-a972-8ca9df4b48df)
Source: Zech et.al. 2014: https://www.dbfz.de/fileadmin/user_upload/Referenzen/DBFZ_Reports/DBFZ_Report_19.pdf